src/citation/views/citation_entry_view.py

```diff
@@ -29,14 +30,18 @@ def user_citations(self, request):
         user = request.user
         organization_id = request.GET.get("organization_id", None)
         project_id = request.GET.get("project_id")
+        citations_query = None
         if project_id:
             citations_query = CitationProject.objects.get(id=project_id).citations.all()
         elif organization_id:
             citations_query = Organization.objects.get(
                 id=organization_id
             ).created_citations.all()
+            if request.query_params.get("get_current_user_citations", None):
+                citations_query = citations_query.filter(created_by=user.id)
```

Comment:
We should do filtering with a `filter_class` instead. See `researchhub_comment/filters.py` for a recent example.